jetstream/cli.py

```diff
@@ -238,38 +239,75 @@ def _delete_enrollment_table(self, experiment: mozanalysis.experiment.Experiment
         enrollments_table = f"{self.project_id}.{self.dataset_id}.enrollments_{normalized_slug}"
         client.delete_table(enrollments_table)
 
-    def _experiments_to_analyse(
+    def _experiments_to_configs(
+        self,
+        experiments: List[mozanalysis.experiment.Experiment],
+        config_getter: Callable[
+            [], ExternalConfigCollection
+        ] = ExternalConfigCollection.from_github_repo,
+    ) -> List[AnalysisConfiguration]:
+        """Convert mozanalysis experiments to analysis configs."""
+        configs = []
+
+        for experiment in experiments:
+            spec = AnalysisSpec.default_for_experiment(experiment)
+            if self.configuration_map and experiment.normandy_slug in self.configuration_map:
+                config_dict = toml.load(self.configuration_map[experiment.normandy_slug])
+                spec.merge(AnalysisSpec.from_dict(config_dict))
+            else:
+                external_configs = config_getter()
+                if external_spec := external_configs.spec_for_experiment(experiment.normandy_slug):
+                    spec.merge(external_spec)
+
+            configs.append(spec.resolve(experiment))
+
+        return configs
+
```

Comment:
Instead of returning the mozanalysis experiment, this will return the AnalysisConfiguration (which will have the right end_date)

Reply:
Can we share this code with SerialExecutorStrategy?

Reply:
We can just pass in the `AnalysisConfiguration`s instead of the experiment slugs into `SerialExecutorStrategy.execute`